### ISSUE TITLE
[windows] Upgrade package-spec

### ISSUE THREAD

**Issue by taylor-swanson** [elastic/integrations#9906]
There are two configuration values flagged in #8610 that need to be marked as secret:

- httpjson.password 
- httpjson.token

In order to support secrets variables and validation, the package-spec version of the integration will need to be raised to 3.0.2 or higher. As a note, migrating to package-spec 3.0 or higher will more than likely trigger validation errors for fields. One of the main reasons why windows is being held back at pkg-spec 1.0.0 is due to the [presence of dynamic metrics fields in the perfmon data stream](https://github.com/elastic/integrations/blob/main/packages/windows/data_stream/perfmon/fields/fields.yml#L12). This is no longer allowed in newer versions of pkg-spec.

Things to do as part of the package-spec bump:

- [ ] Mark `httpjson.password` and `httpjson.token` as `secret: true`
- [ ] Fix `manifest.yml` options
- [ ] Consider using [TSDB](https://github.com/elastic/integrations/blob/main/docs/developer_tsdb_migration_guidelines.md) for `perfmon` data stream
- [ ] Fix field definitions
  - [ ] Fix array type fields ie: `invocation_details`
  - [ ] Fix dynamic metric fields (check [subobjects](https://github.com/elastic/integrations/blob/main/docs/subobjects_adoption_guide.md)
- [ ] Rework dashboards to use Lens

Some of the changes will probably be breaking changes :

- [ ] Update major version
- [ ] Test upgrade process and document any issues given windows is a highly used integration

**Comment by jlind23** [comment]
@cmacknz @pierrehilbert any reason why the data plane team is mentioned as the codeowner here https://github.com/elastic/integrations/blob/ca7078d194b387d0d390e996a09dcddedc84b491/.github/CODEOWNERS#L352C1-L353C1 ?

**Comment by pierrehilbert** [comment]
We discussed with @nfritts a month ago and his team should be the owner.

**Comment by jlind23** [comment]
@pierrehilbert let's make sure we have a pr for this then. @nfritts what's the right github team we should use then?

**Comment by pierrehilbert** [comment]
Yes we should, we are waiting to have a confirmation from @nfritts about the team we should add instead of the Data Plane team to open a PR.

**Comment by nfritts** [comment]
The data plane team seems to own this too: https://github.com/elastic/integrations/blob/ca7078d194b387d0d390e996a09dcddedc84b491/.github/CODEOWNERS#L361C1-L361C72 

I don't know if they should or not?  If they should, then maybe it makes sense for them to remain on the root `windows` folder like they currently are?   If it doesn't make sense then they could just be removed as `sec-windows-platform` is already on it.

**Comment by marc-gr** [comment]
Closed by https://github.com/elastic/integrations/pull/10781
